plugins/kotlin/completion/src/org/jetbrains/kotlin/idea/completion/NamedArgumentCompletion.kt

```diff
@@ -118,10 +116,23 @@ object NamedArgumentCompletion {
 /**
  * Checks whether argument in the [resolvedCall] can be used without its name (as positional argument).
  */
-fun KtValueArgument.canBeUsedWithoutNameInCall(resolvedCall: ResolvedCall<out CallableDescriptor>): Boolean {
+fun KtValueArgument.canBeUsedWithoutNameInCall(
+    resolvedCall: ResolvedCall<out CallableDescriptor>,
+    noOtherNamedArguments: Boolean = false
+): Boolean {
     val valueArguments = resolvedCall.call.valueArguments
+    val parameter = resolvedCall.getParameterForArgument(this)
+    val isVararg = parameter?.isVararg == true
+
+    if (noOtherNamedArguments) {
+        val parameters = resolvedCall.resultingDescriptor.valueParameters
+        if (isVararg && parameter != parameters.lastOrNull()) return false
+        val firstParameter = parameters.firstOrNull()
+        if (firstParameter?.hasDefaultValue() == true && resolvedCall.valueArguments[firstParameter] is DefaultValueArgument) return false
+        return isVararg || isNamed() || placedOnItsOwnPositionInCall(resolvedCall)
+    }
```

Comment:
By asking you to "abstract" the code I didn't mean to introduce one boolean flag to this function which changes this function behavior completely. 

Well, indeed, `noOtherNamedArguments` boolean flag introduction made this code better because now the logic of "argument can be used without name" is localized in one function and not spread around different places in the codebase.

But still, it's not enough. Right now, if anyone wanted to fix any other bug in this subsystem (e.g. [this bug](https://youtrack.jetbrains.com/issue/KTIJ-19737)) then he/she would need to fix the bug twice -- once for "Remove argument name" intention and once for "Remove all argument names" intention.

Let me help you a bit with abstracting this API. Try it this way: change this function signature to: `fun collectAllArgumentsThatCanBeUsedWithoutName(...): List<RemoveNamedArgumentNameTransformation>`

where `RemoveNamedArgumentNameTransformation` is:
```
data class RemoveNamedArgumentNameTransformation(val argumentName: String, val argumentNewIndex: Int)
```

(this API is approximate and only to give you the overall idea)

You can use this API in both "Remove argument name" and "Remove all argument names" intentions.

Abstracting the API this way will fix [this issue](https://youtrack.jetbrains.com/issue/KTIJ-20090) too.